### PR TITLE
No empty strings in parser

### PR DESCRIPTION
**Pull request by keis**
Extracted from #252 

* No more empty strings in parsed array from consecutive whitespace
* Drop unquote option and make it the default behaviour, it was always set to TRUE

### CODE REVIEW

**Comment by mathstuf** [comment]
Yay! We have tests now right? Probably worth a few.

File: src/commands.c [hunk 1]
```diff
@@ -470,39 +470,27 @@ split_quoted (const gchar *src, const gboolean unquote)
...
-            } else {
-                g_string_append_c (str, *p);
-                ctx_single_quote = ! ctx_single_quote;
-            }
+            ctx_single_quote = !ctx_single_quote;
+            ++p;
         } else if (isspace (*p) && !ctx_double_quote && !ctx_single_quote) {
             /* Argument separator. */
-            /* FIXME: Is "a  b" three arguments? */
+            while (isspace(*++p);
```
Comment (mathstuf) [inline src/commands.c]: Hmm. Did this compile?

Comment (keis) [inline src/commands.c]: no :(

small changes right before pushing is a real curse

**Comment by keis** [comment]
We only have the event manager tests suite. I'll take a look to see what it would take to get some tests going for this

File: src/commands.c (hunk 1 above)
Comment (mathstuf) [inline src/commands.c]: Yeah, I know that story :) .

**Comment by keis** [comment]
Getting the tests to work is a bit of hassle as you have to create a `local.mk` that defines `CPPFLAGS=-DUZBL_LIBRARY` so we don't end up with conflicting main methods

**Comment by mathstuf** [comment]
Hmm. We could make a static `libuzbl-core.a` that we don't ship. Once uzbl is multi-threaded, `libuzbl-core.so` could be a thing.

**Comment by keis** [comment]
Creating the lib is not something I would want to have as an extra step to run the tests because it would mean recompiling everything again with some different defines. Anyway, I think I found a decent way of doing these where I exclude uzbl-core from the objects linked with the test runner.

**Comment by mathstuf** [comment]
I meant that `libuzbl-core.a` should always be compiled, even for the `uzbl-core` we ship.

**Comment by keis** [comment]
I was looking at the old makefile bit for creating libuzbl and then we compiled .lo versions of each c file in addition to the .o we were already creating.

I'm not opposed to a libuzbl, but tbh I don't really see the use, as a future goal. When doing that though it would be better if we did not use conditional compilation and maybe even let uzbl-core become a small wrapper around the library.

**Comment by mathstuf** [comment]
Yeah, building only `libuzbl.a` would be fine.

**Comment by keis** [comment]
ok. that's easy enough :)
building a libuzbl.a now!

**Comment by keis** [comment]
ping @mathstuf want to give this another quick look?

**Comment by keis** [comment]
boop

**Comment by mathstuf** [comment]
Hmm. Seems Travis doesn't run the C tests. Or am I missing them?

File: tests/core-tests.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+int
+main (int argc, char *argv[])
+{
+    g_test_init (&argc, &argv, NULL);
+
+    uzbl_commands_init ();
+
+    g_test_add_func ("/uzbl/commands/parse_simple", test_parse_simple);
+    g_test_add_func ("/uzbl/commands/parse_quoted", test_parse_quoted);
+    g_test_add_func ("/uzbl/commands/parse_extra_whitespcae", test_parse_extra_whitespace);
```
Comment (mathstuf) [inline tests/core-tests.c]: "whitespcae" -> "whitespace"

And other than adding `make tests` to the Travis config, LGTM.

**Comment by keis** [comment]
Done. Wrestling travis took a few attempts hehe. Right now the c tests are running for every python version which is a bit unfortunate.

I'll try to find a better way of doing that in a follow up :)

**Pull request merged**